src/FormElement/ScheduleElement/WeeklyFields.php

```diff
@@ -0,0 +1,141 @@
+
+class WeeklyFields extends FieldsetElement
+{
+    use FieldsProtector;
+
+    /** @var array A list of valid week days */
+    protected $weekdays = [];
+
+    /** @var string A valid weekday to be selected by default */
+    protected $default = 'MO';
+
+    protected function init(): void
+    {
+        parent::init();
+
+        $this->weekdays = [
+            'MO' => $this->translate('Mon'),
+            'TU' => $this->translate('Tue'),
+            'WE' => $this->translate('Wed'),
+            'TH' => $this->translate('Thu'),
+            'FR' => $this->translate('Fri'),
+            'SA' => $this->translate('Sat'),
+            'SU' => $this->translate('Sun')
+        ];
+    }
+
+    /**
+     * Set the default weekday to be preselected
+     *
+     * @param string $default
+     *
+     * @return $this
+     */
+    public function setDefault(string $default): self
+    {
+        $weekday = strlen($default) > 2 ? substr($default, 0, -1) : $default;
+        // Attributes are registered far before the initialization of this element!
+        if (! empty($this->weekdays) && ! isset($this->weekdays[strtoupper($weekday)])) {
+            throw new InvalidArgumentException(sprintf('Invalid weekday provided: %s', $default));
+        }
+
+        $this->default = strtoupper($weekday);
+
+        return $this;
+    }
+
+    /**
+     * Get all the selected weekdays
+     *
+     * @return array
+     */
+    public function getSelectedWeekDays(): array
+    {
+        $selectedDays = [];
+        foreach ($this->weekdays as $day => $_) {
+            if ($this->getValue($day, 'n') === 'y') {
+                $selectedDays[] = $day;
+            }
+        }
+
+        if (empty($selectedDays)) {
+            $selectedDays[] = $this->default;
+        }
+
+        return $selectedDays;
+    }
+
+    /**
+     * Transform the given weekdays into key=>value array that can be populated
+     *
+     * @param array $days
+     *
+     * @return array
+     */
+    public function loadWeekDays(array $days): array
+    {
+        $values = [];
+        foreach ($days as $day) {
+            $weekDays = strtoupper($day);
+            if (! isset($this->weekdays[$weekDays])) {
+                throw new InvalidArgumentException(sprintf('Invalid weekday provided: %s', $day));
+            }
+
+            $values[$weekDays] = 'y';
+        }
+
+        return $values;
+    }
+
+    protected function assemble()
+    {
+        $this->getAttributes()->set('id', $this->protectId('weekly-fields'));
+
+        $fieldsWrapper = HtmlElement::create('div', ['class' => 'weekly']);
+        $listItems = HtmlElement::create('ul', ['class' => ['schedule-element-fields', 'multiple-fields']]);
+
+        $foundCheckedDay = false;
+        foreach ($this->weekdays as $day => $value) {
+            $checkbox = $this->createElement('checkbox', $day, [
+                'class' => 'sr-only autosubmit',
```

Comment:
If you define multiple attribute values this way, please use an array. Any subsequent call to `$attribute->add()` then behaves correctly. This is not that important here now, but it's a good habit to remember this and not use a single string for multiple values.